aggregator/aggregator.go

```diff
@@ -86,20 +86,33 @@ func Stdev(in []float64) float64 {
 	return math.Sqrt(variance)
 }
 
-func Sum(in []float64) float64 {
+func Sum(in []float64, ts []uint) float64 {
 	sum := float64(0)
 	for _, term := range in {
 		sum += term
 	}
 	return sum
 }
 
+func Rate(in []float64, ts []uint) float64 {
+	if len(in) == 0 {
+		panic("rate() called in aggregator with 0 terms")
+	}
+	Vnew := in[len(in)-1]
+	Vold := in[0]
+	Tnew := ts[len(ts)-1]
+	Told := ts[0]
+
+	return (Vnew - Vold) / float64(Tnew-Told)
```

Comment:
this could be a division by zero if only 1 point came in. i'm working on implementing this with the new code base and will come up with a solution